frontend/components/suggestion-and-search.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import debounce from 'lodash.debounce';
+import { Badge } from './badge-examples';
+
+function toXhrSend(data) {
+  let str = '', start = true;
+  let jsondata = '';
+  for (const key in data) {
+    if (typeof (jsondata = JSON.stringify(data[key])) === 'string') {
+      str += start ? '' : '&';
+      if (typeof data[key] === 'string') {
+        str += encodeURIComponent(key) + '=' + encodeURIComponent(data[key]);
+      } else {
+        str += encodeURIComponent(key) + '=' + encodeURIComponent(jsondata);
+      }
+      start = false;
+    }
+  }
+  return str;
+}
+
+function ajax(baseUri, verb, adverbs, cb) {
+  const xhr = new XMLHttpRequest();
+  xhr.open('GET', baseUri + "/$" + verb + '?' + toXhrSend(adverbs), true);
+  xhr.onload = e => {
+    if (xhr.readyState === 4) {
```

Comment:
No need to check `readyState` in modern browsers (with XHR2), since this is using `onload`.